RecoLocalTracker/SiPixelRecHits/src/PixelCPEBase.cc

```diff
@@ -239,9 +239,24 @@ PixelCPEBase::setTheClu( DetParam const & theDetParam, ClusterParam & theCluster
    minInY = theClusterParam.theCluster->minPixelCol();
    maxInX = theClusterParam.theCluster->maxPixelRow();
    maxInY = theClusterParam.theCluster->maxPixelCol();
-   
-   theClusterParam.isOnEdge_ = theDetParam.theRecTopol->isItEdgePixelInX(minInX) | theDetParam.theRecTopol->isItEdgePixelInX(maxInX) |
-   theDetParam.theRecTopol->isItEdgePixelInY(minInY) | theDetParam.theRecTopol->isItEdgePixelInY(maxInY) ;
+
+   if      ( theDetParam.theRecTopol->isItEdgePixelInX(minInX) )
+     theClusterParam.edgeTypeX_ = 1;
+   else if ( theDetParam.theRecTopol->isItEdgePixelInX(maxInX) )
+     theClusterParam.edgeTypeX_ = 2;
+   else
+     theClusterParam.edgeTypeX_ = 0;
+     
+   if      ( theDetParam.theRecTopol->isItEdgePixelInY(minInY) )
+     theClusterParam.edgeTypeY_ = 1;
+   else if ( theDetParam.theRecTopol->isItEdgePixelInY(maxInY) )
+     theClusterParam.edgeTypeY_ = 2;
+   else
+     theClusterParam.edgeTypeY_ = 0;
+   
+   theClusterParam.isOnEdge_ = ( theClusterParam.edgeTypeX_ || theClusterParam.edgeTypeY_ );
+   // theClusterParam.isOnEdge_ = theDetParam.theRecTopol->isItEdgePixelInX(minInX) | theDetParam.theRecTopol->isItEdgePixelInX(maxInX) |
```

Comment:
Are these commented out lines needed? If so, please add another line of comment which explain why; otherwise just remove them

Reply:
I wanted to keep the old version around until we are sure we don't need it.  But yes, I can remove the comment now if you require this change now.  (I would have deleted in the future anyway.)